### PR TITLE
add region as config input

### PR DESCRIPTION
**Pull request by robpickerill**
This PR adds the possibility to provide the region for the cloudwatch via the protobuf configuration.

Resolves #671 

Will add docs, do you want docs in separate PR again?

### CODE REVIEW

**Comment by benner** [comment]
I think this can  be auto fetch from existing AWS vars from metadata (https://github.com/google/cloudprober/blob/master/sysvars/sysvars_ec2.go)

image

**Comment by robpickerill** [comment]
Thanks, I never noticed that - would there still be a case whereby the config input would be useful?

Would it be useful if we did:

1. config
2. sysvars
3. env vars (as implemented before this PR)

**Comment by robpickerill** [comment]
@googlebot I fixed it.

**Comment by robpickerill** [comment]
@googlebot I fixed it.

**Comment by manugarg** [comment]
Thanks @robpickerill. I'll start the process of importing this PR.

File: surfacers/cloudwatch/cloudwatch.go [hunk 1]
```diff
@@ -154,13 +155,34 @@ func emLabelsToDimensions(em *metrics.EventMetrics) []*cloudwatch.Dimension {
 	return dimensions
 }
 
+// getRegion provides an order of precedence for the lookup of the AWS Region
+// Precendence:
+// 1. The region passed in via config
+// 2. The region discovered from the metadata endpoint
+// 3. By returning nil, the environment variable AWS_REGION as evaluated by the
+// the AWS SDK.
+func getRegion(config *configpb.SurfacerConf) *string {
```
Comment (manugarg) [inline surfacers/cloudwatch/cloudwatch.go:164]: I think it's better to return a string here and do the allocation in New(). Returning a pointer here could be a little tricky.

**Review by manugarg — COMMENTED** [review]
Hey Rob,

Submitting this change through: https://github.com/google/cloudprober/pull/672. I made one small change mentioned below. Hope that's okay.

File: surfacers/cloudwatch/cloudwatch.go (hunk 1 above)
Comment (robpickerill) [reply surfacers/cloudwatch/cloudwatch.go:164]: No problems, agree and sounds good to me.

**Comment by manugarg** [comment]
Cool. Closing this.

**Pull request closed without merging**